serve/mlc_serve/engine/staging_engine_worker.py

```diff
@@ -114,8 +112,12 @@ def add(self, request_states: list[RequestState]):
                 if (
                     request_state.validation_err is not None
                     or request_state.prompt_len >= self.max_context_length
+                    or request_state.prompt_len > self.max_num_batched_tokens
+                    or self.cache_manager.get_kv_cache_size() - request_state.prompt_len < self.max_decode_steps
```

Comment:
Document why we need this condition. My understanding is that if we remove this condition and allow such request, the request can never put into the batch due to the condition https://github.com/octoml/mlc-llm/blob/04a27a0f09e7a43608f22e1c99089da390452ddc/serve/mlc_serve/engine/staging_engine_worker.py#L279-L284

Reply:
exactly, that will lead to hang. To avoid such situation, we added verification on request add stage and cancel it if we understand that we cannot add such request for processing